pages/tech-partners/index.jsx

```diff
@@ -0,0 +1,344 @@
+    acc[post.slug] = post._allReferencingTechPartners;
+    return acc;
+  }, {});
+
+  const highlightedSlugs = techPartnersPage.highlightedPartners.map(
+    (p) => p.slug,
+  );
+
+  const ordered = [
+    ...techPartnersPage.highlightedPartners,
+    ...sortBy(
+      posts.filter((p) => !highlightedSlugs.includes(p.slug)),
+      [(x) => -(countBySlug[x.slug] || 0), 'slug'],
+    ),
+  ];
+
+  const [technologyFilter, setTechnologyFilter] = useState(null);
+  const [areaOfExpertiseFilter, setAreaOfExpertiseFilter] = useState(null);
+  const [countryFilter, setCountryFilter] = useState(null);
+  const [continentFilter, setContinentFilter] = useState(null);
+
+  const filtered = ordered.filter((p) => {
+    if (
+      technologyFilter &&
+      !p.technologies.some((t) => t.name === technologyFilter)
+    ) {
+      return false;
+    }
+    if (
+      areaOfExpertiseFilter &&
+      !p.areasOfExpertise.some((t) => t.name === areaOfExpertiseFilter)
+    ) {
+      return false;
+    }
+    if (
+      countryFilter &&
+      !p.locations.some((t) => `${t.emoji} ${t.name}` === countryFilter)
+    ) {
+      return false;
+    }
+    if (
+      continentFilter &&
+      !p.locations.some((t) => t.continent.name === continentFilter)
+    ) {
+      return false;
+    }
+
+    return true;
+  });
+
+  const filteredCounters = calculateCounters(filtered, continentFilter);
+  const noFilterCounters = calculateCounters(ordered, continentFilter);
+
+  const someFilters =
+    continentFilter ||
+    countryFilter ||
+    technologyFilter ||
+    areaOfExpertiseFilter;
+
+  const continentOptions = Object.entries(filteredCounters.continents)
+    .sort(byCount)
+    .map(toOption);
+
+  const countryOptions = Object.entries(filteredCounters.countries)
+    .sort(byCount)
+    .map(toOption);
+
+  const technologyOptions = Object.entries(filteredCounters.technologies)
+    .sort(byCount)
+    .map(toOption);
+
+  const areaOfExpertiseOptions = Object.entries(
+    filteredCounters.areaOfExpertise,
+  )
+    .sort(byCount)
+    .map(toOption);
+
+  return (
+    <Layout preview={preview}>
+      <Head>
+        <title>DatoCMS Technology Partners</title>
+      </Head>
+      <Wrapper>
+        <Hero
+          kicker="Ecosystem Partners"
+          title={
+            <>
+              Find the perfect Tech Partner to supercharge{' '}
+              <Highlight>your DatoCMS projects</Highlight>
+            </>
+          }
+          subtitle={
+            <>
+              DatoCMS partners with leading technology providers and APIs to
+              ensure your projects are getting the best possible treatment.
+            </>
+          }
+        />
+
+        <Space bottom={1}>
+          <Announce href="/partner-program" center>
+            <strong>Want to become a DatoCMS Partner?</strong> Learn more about
+            our Partner Program and its benefits!
+          </Announce>
+        </Space>
+
+        {/* <div className={s.filterGrid}>
```

Comment:
This area is commented because it might be useful when the number of tech partners is higher, right?